src/main/java/way/Route.java

```diff
@@ -46,7 +58,13 @@ public Route(String name, double mileageBegin, double mileageEnd, TrafficInfo tr
         this.spacesByDirection = new HashMap<>();
         String dir1;
         String dir2;
-        if (!name.equals("") && Integer.parseInt(name.replaceAll("[\\D]", "")) % 2 == 0) {
+        int number = 0;
+        try {
+            number = makeRoadNumber(name);
```

Comment:
Ciągle nie wiem czemu tu jest walidacja. Konstruktor nie wydaje mi się być odpowiednim miejscem na to, lepiej się upewniać że do konstruktora dajemy już poprawne. W jakim przypadku nie możemy tak zrobić?

Reply:
no to jest już poprawne. Tylko nie chce mi się pisać drugiej funkcji, która będzie robiła to samo, a nie rzucała wyjątku, dlatego używam makeRoadNumber (ponownie)

Reply:
A, to spoko